library/src/utils/isLuhnAlgo/isLuhnAlgo.ts

```diff
@@ -7,7 +12,7 @@
  */
 export function isLuhnAlgo(input: string) {
   // Remove any non-digit chars
-  const number = input.replace(/\D/gu, '');
+  const number = input.replace(NON_DIGIT_REGEX, '');
```

Comment:
needs `replaceAll` here (could be automatically captured by https://github.com/fabian-hiller/valibot/pull/213/files `unicorn/prefer-string-replace-all` I think.

reason is: a global regex (`/g`) will set `lastIndex` to the last index where the regex stopped. Which means if string a is "abcdef" and string b is "a", lastIndex will be set to e.g. 3, but string b has no index 3, so the regex now fails at all times.
If you want to avoid `replaceAll`, you'll need to call
`NON_DIGIT_REGEX.lastIndex = 0` before calling the replace